src/main/java/org/gaul/s3proxy/junit/S3ProxyRule.java

```diff
@@ -45,60 +34,44 @@
     private static final Logger logger = LoggerFactory.getLogger(
         S3ProxyRule.class);
 
-    private static final String LOCALHOST = "127.0.0.1";
-
-    private final String accessKey;
-    private final String secretKey;
-    private final String endpointFormat;
-    private final S3Proxy s3Proxy;
-
-    private final BlobStoreContext blobStoreContext;
-    private URI endpointUri;
-    private final File blobStoreLocation;
+    private final S3ProxyJunitCore core;
 
     public static final class Builder {
-        private AuthenticationType authType = AuthenticationType.NONE;
-        private String accessKey;
-        private String secretKey;
-        private String secretStorePath;
-        private String secretStorePassword;
-        private int port = -1;
-        private boolean ignoreUnknownHeaders;
-        private String blobStoreProvider = "filesystem";
 
-        private Builder() { }
+        private final S3ProxyJunitCore.Builder builder;
+
+        private Builder() {
+            builder = new S3ProxyJunitCore.Builder();
+        }
 
         public Builder withCredentials(AuthenticationType authType,
-                String accessKey, String secretKey) {
-            this.authType = authType;
-            this.accessKey = accessKey;
-            this.secretKey = secretKey;
+                                                        String accessKey, String secretKey) {
+            builder.withCredentials(authType, accessKey, secretKey);
             return this;
         }
 
         public Builder withCredentials(String accessKey, String secretKey) {
             return withCredentials(AuthenticationType.AWS_V2_OR_V4, accessKey,
```

Comment:
```suggestion
          builder.withCredentials(accessKey, secretKey);
          return this;
```
(as per previous comment in the extension).